variant_remapping_tools/reads_to_remapped_variants.py

```diff
@@ -7,75 +7,66 @@
-
-
-def is_read_valid(read, counter, flank_length, score_cutoff, diff_cutoff):
-    """
-    Filter out reads that are:
-        - Unmapped
-        - Non primary (secondary): AS's that are too low and XS's that are too close to AS
-    """
-    if read.is_secondary:  # We only want to deal with primary reads
-        return False
-    counter['total'] += 1
-    if read.is_unmapped:  # Can be decoded with bitwise flag with & 4 (4 means unmapped)
-        counter['unmapped'] += 1
-        return False
-    # Getting AS and XS:
-    AS = read.get_tag('AS')
-    try:
-        XS = read.get_tag('XS')  # Some variants don't have secondary alignments, which throws an error
-    except KeyError:
-        XS = -flank_length  # Set an arbitrary low value for the 'artificial' secondary alignment
-    if AS <= int(score_cutoff):
-        counter['primary_poor'] += 1
-        return False
-    if AS - XS < diff_cutoff:
-        counter['gap_small'] += 1
-        return False
-    return True
-
-
-def is_local_alignment_perfect(counter, perf_counter, local_region_size):
+def reverse_complement(sequence):
+    return str(Seq(sequence, generic_dna).reverse_complement())
+
+
+def calculate_new_variant_definition(left_read, right_read, ref_fasta):
     """
-    Filter out non-perfect local alignment around the variant
+    Resolve the variant definition from the flanking region alignment and old variant definition
+    TODO: Link to algorithm description once public
     """
-    if perf_counter != 2 * local_region_size:
-        counter['context_bad'] += 1
-        return False
-    # Filter out AS's that are too low and XS's that are too close to AS, and those with a non-perfect local alignment:
-    counter['remapped'] += 1
-    return True
+    # grab information from the read name
+    name = left_read.query_name
+    info = name.split('|')
+    old_ref = info[2]
+    old_alts = info[3].split(',')
+
+    # Define new ref and new pos
+    new_ref = fetch_bases(ref_fasta, left_read.reference_name, left_read.reference_end + 1,
+                          right_read.pos - left_read.reference_end)
+    new_pos = left_read.reference_end + 1
+
+    # 1. Handle reference strand change
+    if not left_read.is_reverse and not right_read.is_reverse:
+        # Forward strand alignment
+        old_ref_conv = old_ref
+        old_alt_conv = old_alts
+
+    elif left_read.is_reverse and right_read.is_reverse:
+        # Reverse strand alignment
+        old_ref_conv = reverse_complement(old_ref)
+        old_alt_conv = [reverse_complement(alt) for alt in old_alts]
+    else:
+        error_msg = (f'Impossible read configuration: '
+                     f'read1 is_reverse: {left_read.is_reverse}, '
+                     f'read2 is_reverse: {right_read.is_reverse}, '
+                     f'read1 position: {left_read.pos}, '
+                     f'read2 position: {right_read.pos}')
+        print(error_msg)
+        raise ValueError(error_msg)
```

Comment:
I actually imagine that such scenarios (read being not properly paired) will occur from time to time in real life situations, either due to aligner error or genuine genome reassembly, and will not always warrant throwing an exception. Perhaps just counting them as one of the modes of failure to remap would be more appropriate?

Reply:
Exactly right.